### ISSUE TITLE
[Reveal] iOS Voiceover accessibility issue

### ISSUE THREAD

**Issue by spiderzaur** [zurb/foundation-sites#10735]
Having issue with Reveal accessibility on iOS. Voiceover does not focus on Reveal modal, instead focus left behind the reveal. This only happens on iOS, it is fine on Android and Desktop.

#### How to reproduce this bug:

1. Enable Voiceover on iOS
2. Open Safari
3. Visit any page has Reveal modal, even official docs page https://foundation.zurb.com/sites/docs/reveal.html
4. Double tap on the button to open the Modal
5. If you swipe left or swipe right (action for Voiceover to read next or previous item), you will hear it is reading the items behind the the Reveal instead of in the Reveal.

#### What should happen: 
Voiceover should focus content inside the reveal modal, not behind it which is not visible to user anymore because Reveal window is open

#### What happened instead:
Voiceover kept focus behind the reveal window and continued reading items behind it  

#### Browser(s) and Device(s) tested on:
iPhone 7 Plus, iOS 11.0.3, Safari

#### Foundation Version(s) you are using:
6.4.3

#### Test case link:

- [CodePen with Foundation 6.3.1, Float Grid and MotionUI](http://codepen.io/IamManchanda/pen/LWGZxR)
- [CodePen with Foundation 6.3.1, Float Grid with RTL Direction and MotionUI](http://codepen.io/IamManchanda/pen/bRYOMv)
- [CodePen with Foundation 6.3.1, Flexbox grid and MotionUI](http://codepen.io/IamManchanda/pen/zZrBEv)

- [CodePen with Foundation 6.4.1, XY Flexbox grid and MotionUI](http://codepen.io/IamManchanda/pen/EXbGKJ)
- [CodePen with Foundation 6.4.1, XY Flexbox grid with RTL Direction and MotionUI](http://codepen.io/IamManchanda/pen/qjVLoO)
- [CodePen with Foundation 6.4.1, XY Flexbox grid with Prototype Mode and MotionUI](http://codepen.io/IamManchanda/pen/XgzopG)
- [CodePen with Foundation 6.4.1, Float Grid and MotionUI](http://codepen.io/IamManchanda/pen/qjVLrB)
